python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -99,6 +100,43 @@ def __post_init__(self) -> None:
             raise TypeError("pass_read_limit must be an int")
 
 
+def default_blocksize(scheduler: str) -> int:
+    """Return the default blocksize."""
+    try:
+        # Use PyNVML to find the worker device size.
+        import pynvml
+
+        pynvml.nvmlInit()
+        index = os.environ.get("CUDA_VISIBLE_DEVICES", "0").split(",")[0]
+        if index and not index.isnumeric():  # pragma: no cover
+            # This means device_index is UUID.
+            # This works for both MIG and non-MIG device UUIDs.
+            handle = pynvml.nvmlDeviceGetHandleByUUID(str.encode(index))
+            if pynvml.nvmlDeviceIsMigDeviceHandle(handle):
+                # Additionally get parent device handle
+                # if the device itself is a MIG instance
+                handle = pynvml.nvmlDeviceGetDeviceHandleFromMigDeviceHandle(handle)
+        else:
+            handle = pynvml.nvmlDeviceGetHandleByIndex(int(index))
+
+        device_size = pynvml.nvmlDeviceGetMemoryInfo(handle).total
+
+    except (ValueError, pynvml.NVMLError):  # pragma: no cover
+        # Fall back to a conservative 8GiB default
+        device_size = 8 * 1024**3
+
+    if scheduler == "distributed":
+        # Distributed execution requires a conservative
+        # blocksize for now.
+        blocksize = int(device_size * 0.025)
+    else:
+        # Single-GPU execution can lean on UVM to
+        # support a much larger blocksize.
+        blocksize = int(device_size * 0.0625)
+
+    return max(blocksize, 256_000_000)
```

Comment:
I think for better performance we had to run in the range of 1-2B, here it's being limited to 256M regardless of device size, is that intentional?

Reply:
This prevents the default blocksize from going **below** 256MB. It can always be larger than that (and the user can explicitly set something lower if they want). Do you think this a problem?